snap/info_snap_yaml.go

```diff
@@ -120,6 +131,34 @@ func InfoFromSnapYaml(yamlData []byte) (*Info, error) {
 	// Bind unbound slots to all apps
 	bindUnboundSlots(globalSlotNames, snap)
 
+	// Collect layout elements.
+	if y.Layout != nil {
+		snap.Layout = make(map[string]*Layout, len(y.Layout))
+		for path, l := range y.Layout {
+			mode := 0755
+			if l.Mode != "" {
+				m, err := strconv.ParseInt(l.Mode, 8, 32)
```

Comment:
`mode_t` is unsigned, and `os.FileMode` is a `uint32`; `math.MaxUint32` is a valid `os.FileMode`, and would fail to parse here. But also, `os.ModeDir`, a constant pre-defined probably-not-that-rare file mode, is not a valid `int32`.